#!/usr/bin/env python
PACKAGE = "gazebo_mecanum_plugins"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("LEFT_REAR_P", double_t, 0, "LEFT_REAR wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("LEFT_REAR_I", double_t, 0, "LEFT_REAR wheel PID Controller I parameter",    0, 0, 1000)
gen.add("LEFT_REAR_D", double_t, 0, "LEFT_REAR wheel PID Controller D parameter",    0, 0, 1000)

gen.add("LEFT_FRONT_P", double_t, 0, "LEFT_FRONT wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("LEFT_FRONT_I", double_t, 0, "LEFT_FRONT wheel PID Controller I parameter",    0, 0, 1000)
gen.add("LEFT_FRONT_D", double_t, 0, "LEFT_FRONT wheel PID Controller D parameter",    0, 0, 1000)

gen.add("RIGHT_FRONT_P", double_t, 0, "RIGHT_FRONT wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("RIGHT_FRONT_I", double_t, 0, "RIGHT_FRONT wheel PID Controller I parameter",    0, 0, 1000)
gen.add("RIGHT_FRONT_D", double_t, 0, "RIGHT_FRONT wheel PID Controller D parameter",    0, 0, 1000)

gen.add("RIGHT_REAR_P", double_t, 0, "RIGHT_REAR wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("RIGHT_REAR_I", double_t, 0, "RIGHT_REAR wheel PID Controller I parameter",    0, 0, 1000)
gen.add("RIGHT_REAR_D", double_t, 0, "RIGHT_REAR wheel PID Controller D parameter",    0, 0, 1000)

exit(gen.generate(PACKAGE, "gazebo_mecanum_plugins", "gazebo_mecanum_plugins_pid"))



